tests/Phoenix.MarketData.Infrastructure.Tests/generate-coverage-report.sh

```
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+# Ensure ReportGenerator tool is installed
+if ! command -v reportgenerator &> /dev/null; then
+    echo "Installing ReportGenerator tool..."
+    dotnet tool install -g dotnet-reportgenerator-globaltool
+fi
+
+# Set working directory to the project directory
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+cd "$SCRIPT_DIR"
```

Shellcheck (0.10.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)